content/blog/_posts/2021-04-rust-custom-test-harness.md

```diff
@@ -0,0 +1,296 @@
+---
+title: How to Build a Custom Test Harness in Rust
+author:
+    name: "T.J. Telan"
+    github: "tjtelan"
+description: "Building a custom test harness in Rust is less complicated following these patterns"
+date: 2021-04-26
```

Comment:
Maybe better title would be `Use Plugin-registry to build custom Test runner in Rust`

Reply:
I was trying to aim for the kind of search query result someone might search for after reading the cargo test man page.

---

From the [man page](https://doc.rust-lang.org/cargo/commands/cargo-test.html):

> The libtest harness may be disabled by setting harness = false in the target manifest settings, in which case your code will need to provide its own main function to handle running tests.

This sentence is very consequential, and I wanted a pattern to copy from where the tests didn't need to do any environment validation.